Service/AlgoliaConnector.php

```diff
@@ -473,122 +500,139 @@ protected static function setLastOperationInfo(string $indexName, array $respons
 
-        self::setLastOperationInfo($indexName, $res, $storeId);
+        $res = $this->getClient($indexOptions->getStoreId())->saveRules($indexName, $rules, $forwardToReplicas);
+
+        $this->setLastOperationInfo($indexOptions, $res);
     }
 
 
     /**
-     * @param string $indexName
+     * @param IndexOptionsInterface $indexOptions
      * @param string $objectID
      * @param bool $forwardToReplicas
      * @return void
      * @throws AlgoliaException
+     * @throws NoSuchEntityException
      */
     public function deleteRule(
-        string $indexName,
+        IndexOptionsInterface $indexOptions,
         string $objectID,
-        bool $forwardToReplicas = false,
-        ?int $storeId = null
+        bool $forwardToReplicas = false
     ): void
     {
-        $res = $this->getClient($storeId)->deleteRule($indexName, $objectID, $forwardToReplicas);
+        $indexName = $this->getIndexName($indexOptions);
 
-        self::setLastOperationInfo($indexName, $res, $storeId);
+        $res = $this->getClient($indexOptions->getStoreId())->deleteRule($indexName, $objectID, $forwardToReplicas);
+
+        $this->setLastOperationInfo($indexOptions, $res);
     }
 
     /**
-     * @param string $fromIndexName
-     * @param string $toIndexName
-     * @param int|null $storeId
+     * @param IndexOptionsInterface $fromIndexOptions
+     * @param IndexOptionsInterface $toIndexOptions
      * @return void
      * @throws AlgoliaException
+     * @throws NoSuchEntityException
      */
-    public function copySynonyms(string $fromIndexName, string $toIndexName, ?int $storeId = null): void
+    public function copySynonyms(IndexOptionsInterface $fromIndexOptions, IndexOptionsInterface $toIndexOptions): void
     {
-        $response = $this->getClient($storeId)->operationIndex(
+        $fromIndexName = $this->getIndexName($fromIndexOptions);
+        $toIndexName = $this->getIndexName($toIndexOptions);
+
+        $response = $this->getClient($fromIndexOptions->getStoreId())->operationIndex(
             $fromIndexName,
             [
                 'operation'   => 'copy',
                 'destination' => $toIndexName,
                 'scope'       => ['synonyms']
             ]
         );
-        self::setLastOperationInfo($fromIndexName, $response, $storeId);
+        $this->setLastOperationInfo($fromIndexOptions, $response);
     }
 
     /**
-     * @param string $fromIndexName
-     * @param string $toIndexName
-     * @param int|null $storeId
+     * @param IndexOptionsInterface $fromIndexOptions
+     * @param IndexOptionsInterface $toIndexOptions
      * @return void
      * @throws AlgoliaException
+     * @throws NoSuchEntityException
      */
-    public function copyQueryRules(string $fromIndexName, string $toIndexName, ?int $storeId = null): void
+    public function copyQueryRules(IndexOptionsInterface $fromIndexOptions, IndexOptionsInterface $toIndexOptions): void
```

Comment:
I'm wondering if we should call out via a doc block that copy operations cannot be performed _across_ two different applications?